core/src/main/java/org/apache/iceberg/avro/AvroFileAppender.java

```diff
@@ -95,10 +95,9 @@ public void close() throws IOException {
 
   @SuppressWarnings("unchecked")
   private static <D> DataFileWriter<D> newAvroWriter(
-      Schema schema, PositionOutputStream stream, DatumWriter<?> metricsAwareDatumWriter,
+      Schema schema, PositionOutputStream stream, DatumWriter<?> datumWriter,
       CodecFactory codec, Map<String, String> metadata) throws IOException {
-    DataFileWriter<D> writer = new DataFileWriter<>(
-        (DatumWriter<D>) metricsAwareDatumWriter);
+    DataFileWriter<D> writer = new DataFileWriter<>((DatumWriter<D>) datumWriter);
```

Comment:
Is this rename needed? While I support cleaning up names, I would generally opt to leave these as-is to have smaller commits that are less likely to cause conflicts.